### PR TITLE
"All done!" screen

### LINKED ISSUE #2753: "All Done!" Screen in Add Dataset Flow
## Description

At the end of the Add Dataset flow, it'd be nice to show the user an "All Done" to show the user that the process has definitively ended and make them feel nice about all the hard work they just did.

## Design
image

https://projects.invisionapp.com/share/KMVQ7DH652J#/screens/402485649

### Changes from Design
The design is pretty heavily predicated on the dataset having gone into a draft state, which we don't have yet. For now:
- "Your dataset has been successfully added to the catalog" should replace "Your dataset has successfully been sent off for approval"
- Remove the "You can view the status of the dataset from your homepage" text
- The upper button should just say "View your dataset"

## Acceptance Criteria
-   **Unless the Add Dataset preview feature flag is set, when the user sees this screen, their dataset should be published as part of the catalog** (until we add a proper draft approval flow soon :tm:)
- The upper button should take the user to the dataset page
- The lower button should also take the user to the dataset page, but immediately trigger the print dialog

## Out of Scope
-    Anything to do with drafts

### PR DESCRIPTION
**Pull request by sajidanower23**
### What this PR does

Fixes #2753 

Uses the [new design](https://projects.invisionapp.com/share/KMVQ7DH652J#/screens/402485649) for the end preview page.

### Checklist

-   [x] ~There are unit tests to verify my changes are correct~ Unit tests aren't applicable
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've linked this PR to an issue in ZenHub (core dev team only)
-   [x] Resize the buttons
-   [x] Add more spacing between text and buttons
-   [x] Remove the `Save and Exit` button
-   [x] Center the elements
-   [x] Remove the "Review files" button
-   [x] Remove the header(s)
-   [x] Add links to the buttons

Here's what it looks like at the moment:
image

### CODE REVIEW

**Comment by sajidanower23** [comment]
I have made a deployment to https://2753-all-done-screen.dev.magda.io if anyone wants to look at it.

**Comment by alyce** [comment]
@sajidanower23 I'm trying to take a look but I can't seem to see it :( Can you add a screenshot or something?

**Comment by sajidanower23** [comment]
@alyce You should be able to see it here: https://2753-all-done-screen.dev.magda.io/dataset/add/metadata/magda-ds-caabf690-732f-40f8-851d-7e775bfc3785/4

Make sure you are sign in at https://2753-all-done-screen.dev.magda.io first!

File: magda-web-client/src/Components/Dataset/Add/Pages/DatasetAddEndPage.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+                        Your dataset has been successfully sent off for
+                        approval.
+                    </h3>
+                    <br />
+                    <p className="dataset-status-txt">
+                        You can view the status of your dataset from{" "}
+                        <a href="/">your home page</a>.
+                    </p>
+                </div>
+                <br />
```
Comment (AlexGilleran) [inline magda-web-client/src/Components/Dataset/Add/Pages/DatasetAddEndPage.tsx:41]: Could you use a margin or padding css style for this? Doing layout like this was pretty common years ago, but it tends to get you in trouble if anyone changes the line-height.

File: magda-web-client/src/Components/Dataset/DatasetPage.tsx [hunk 2]
```diff
@@ -169,6 +169,14 @@ const DatasetPage: FunctionComponent<PropsType> = props => {
             </div>
         </div>
     );
+
+    useEffect(() => {
+        if (props.history.location.search === "?print=true") {
```
Comment (AlexGilleran) [inline magda-web-client/src/Components/Dataset/DatasetPage.tsx:174]: Might want to use `query-string` for this (see the Account component for an example).

The problem with doing it like this is that if anyone every includes another query parameter and puts it first (e.g. `?foo=bar&print=true` this won't work anymore.

File: magda-web-client/src/Components/Dataset/Add/Pages/DatasetAddEndPage.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+type Props = {
+    datasetId: string;
+};
+
+// If you are not in preview mode
+export default function DatasetAddEndPage(props: Props) {
+    const draftPage = "/dataset/" + props.datasetId + "/details";
+    const viewDraft = () => {
+        window.location.href = draftPage;
```
[the remark this answers is not in the document]
Comment (sajidanower23) [reply magda-web-client/src/Components/Dataset/Add/Pages/DatasetAddEndPage.tsx]: Good point :+1:

File: magda-web-client/src/Components/Dataset/Add/Pages/DatasetAddEndPage.tsx (hunk 1 above)
Comment (sajidanower23) [reply magda-web-client/src/Components/Dataset/Add/Pages/DatasetAddEndPage.tsx]: Styled with `margin-top`

**Pull request merged**